clang/lib/Frontend/CompilerInstance.cpp

```diff
@@ -1044,6 +1044,9 @@ bool CompilerInstance::ExecuteAction(FrontendAction &Act) {
     llvm::EnableStatistics(false);
 
   for (const FrontendInputFile &FIF : getFrontendOpts().Inputs) {
+    std::string filename = FIF.getFile().str();
+    llvm::PrettyStackTraceFormat trace("Acting on file '%s'", filename.c_str());
```

Comment:
Nit: to avoid an allocation, could you use `%.*` and pass in the length?  It already gets copied internally, so the lifetime of what you pass in doesn't matter (although FIF.File is probably not modified here regardless).

 Untested:
```c++
llvm::PrettyStackTraceFormat trace("Acting on file '%.*s'", FIF.getFile().size(), FIF.getFile().data());
```